packages/amplication-git-utils/src/utils/prepare-files-for-pull-request.ts

```diff
@@ -0,0 +1,82 @@
+import { join } from "path";
+import { Changes } from "octokit-plugin-create-pull-request/dist-types/types";
+import { AMPLICATION_IGNORED_FOLDER } from "../git/git.constants";
+import { AmplicationIgnoreManger } from "./amplication-ignore-manger";
+
+export async function prepareFilesForPullRequest(
+  owner,
+  repositoryName,
+  gitResourceMeta,
+  pullRequestModule
+): Promise<Required<Changes["files"]>> {
+  const amplicationIgnoreManger = new AmplicationIgnoreManger();
+  await amplicationIgnoreManger.init(async (fileName) => {
+    try {
+      const file = await this.getFile({
+        owner,
+        repositoryUrl: repositoryName,
+        path: fileName,
+        baseBranch: undefined, // take the default branch
+      });
+      const { content, htmlUrl, name } = file;
+      console.log(`Got ${name} file ${htmlUrl}`);
+      return content;
+    } catch (error) {
+      console.log("Repository does not have a .amplicationignore file");
+      return "";
+    }
+  });
+
+  //do not override files in 'server/src/[entity]/[entity].[controller/resolver/service/module].ts'
+  //do not override server/scripts/customSeed.ts
+  const doNotOverride = [
```

Comment:
I'm not sure you need `gitResourceMeta.serverPath` for those regex and if you don't, it's better to set the regex outside of the function so it wont set it every time

Reply:
this is an existing logic that I moved/refactored